android/src/main/java/com/bitmovin/player/reactnative/BitmovinBaseModule.kt

```diff
@@ -0,0 +1,71 @@
+package com.bitmovin.player.reactnative
+
+import com.bitmovin.player.api.Player
+import com.bitmovin.player.reactnative.extensions.drmModule
+import com.bitmovin.player.reactnative.extensions.offlineModule
+import com.bitmovin.player.reactnative.extensions.playerModule
+import com.bitmovin.player.reactnative.extensions.sourceModule
+import com.bitmovin.player.reactnative.extensions.uiManagerModule
+import com.facebook.react.bridge.*
+import com.facebook.react.uimanager.UIManagerModule
+
+/**
+ * Base for Bitmovin React modules.
+ *
+ * Provides many helper methods that are promise exception safe.
+ *
+ * In general, code should not throw while resolving a [Promise]. Instead, [Promise.reject] should be used.
+ * This doesn't match Kotlin's error style, which uses exception. The helper methods in this class, provide such
+ * convenience, they can only be called in a context that will catch any Exception and reject the [Promise].
+ *
+ */
+abstract class BitmovinBaseModule(
+    protected val context: ReactApplicationContext,
+) : ReactContextBaseJavaModule(context) {
+    /** [resolve] the [Promise] by running [block] in the UI thread with [UIManagerModule.addUIBlock].  */
+    protected inline fun <T> Promise.resolveOnUIThread(crossinline block: RejectPromiseOnExceptionBlock.() -> T) {
+        val uiManager = runAndRejectOnException { uiManager } ?: return
+        uiManager.addUIBlock {
+            resolveOnCurrentThread {
+                resolve(block())
```

Comment:
I think `resolveOnCurrentThread` already takes care of resolving.
```suggestion
                block()
```

Reply:
It's fine, it should create an error on the react side, and should be covered by the soon existing system tests as well.